pwndbg/gdblib/disasm/arch.py

```diff
@@ -721,5 +722,54 @@ def _memory_string(self, instruction: PwndbgInstruction, operand: EnhancedOperan
         else:
             return None
 
+    def _common_cmp_annotator_builder(
+        self, flags_register_name: str, char_to_separate_operands: str = "-"
+    ) -> Callable[[PwndbgInstruction, Emulator], None]:
+        """
+        Many architectures implement near-identical `CMP`-like instructions.
+
+        It takes two values, either subtracts, adds, or does some bit operation
+        with them to set values in the flag register.
+
+        To reduce code duplication, subclasses can use this function to create an annotator for CMP-like instructions.
+        """
+        SPACES = 5
+        FLAG_REG_NAME_DISPLAY = flags_register_name.upper()
+
+        def handler(instruction: PwndbgInstruction, emu: Emulator):
+            # If there are just two operands, we can assume we are comparing them directly, and can display the values.
+            # Some architectures have variants with more operands that modify the values before comparison.
+            if len(instruction.operands) == 2:
+                left, right = instruction.operands
+
+                if (
+                    left.before_value_resolved is not None
+                    and right.before_value_resolved is not None
+                ):
+                    print_left, print_right = pwndbg.enhance.format_small_int_pair(
+                        left.before_value_resolved, right.before_value_resolved
```

Comment:
We should probably assign those two vars in the `if` with `:=`?

Reply:
Done